OutOfSchool/OutOfSchool.WebApi/Controllers/V1/AreaAdminController.cs

```diff
@@ -0,0 +1,350 @@
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using OutOfSchool.Common.Models;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Models;
+using OutOfSchool.WebApi.Common;
+using OutOfSchool.WebApi.Enums;
+using OutOfSchool.WebApi.Models;
+
+namespace OutOfSchool.WebApi.Controllers.V1;
+
+[ApiController]
+[ApiVersion("1.0")]
+[Route("api/v{version:apiVersion}/[controller]/[action]")]
+public class AreaAdminController : Controller
+{
+    private readonly IAreaAdminService areaAdminService;
+    private readonly ILogger<AreaAdminController> logger;
+    private string path;
+    private string currentUserId;
+    private string currentUserRole;
+
+    public AreaAdminController(
+        IAreaAdminService areaAdminService,
+        ILogger<AreaAdminController> logger)
+    {
+        ArgumentNullException.ThrowIfNull(areaAdminService);
+        ArgumentNullException.ThrowIfNull(logger);
+
+        this.areaAdminService = areaAdminService;
+        this.logger = logger;
+    }
+
+    public override void OnActionExecuting(ActionExecutingContext context)
+    {
+        path = $"{context.HttpContext.Request.Path.Value}[{context.HttpContext.Request.Method}]";
+        currentUserId = GettingUserProperties.GetUserId(User);
+        currentUserRole = GettingUserProperties.GetUserRole(User);
+    }
+
+    /// <summary>
+    /// To Get the Profile of authorized AreaAdmin.
+    /// </summary>
+    /// <returns>Authorized AreaAdmin's profile.</returns>
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(RegionAdminDto))]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [HasPermission(Permissions.PersonalInfo)]
+    [HttpGet]
+    public async Task<IActionResult> Profile()
+    {
+        if (currentUserId == null)
+        {
+            return BadRequest("Invalid user information.");
+        }
+
+        try
+        {
+            var areaAdmin = await areaAdminService.GetByUserId(currentUserId).ConfigureAwait(false);
+            return Ok(areaAdmin);
+        }
+        catch (ArgumentException e)
+        {
+            return NotFound(e.Message);
+        }
+    }
+
+    /// <summary>
+    /// Get AreaAdmin by it's id.
+    /// </summary>
+    /// <param name="id">AreaAdmin id.</param>
+    /// <returns>Authorized AreaAdmin's profile.</returns>
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(RegionAdminDto))]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status403Forbidden)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [HasPermission(Permissions.RegionAdminRead)]
```

Comment:
I think all actions in this controller should be marked by AreaAdmin* permissions, and the other administrators should have these necessary permissions in accordance with the Roles and Permissions Matrix to manipulate the area admins.